src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -3853,12 +3849,27 @@ public function remove_class( $class_name ): bool {
 		}
 
 		if ( null !== $this->tag_name_starts_at ) {
-			$this->classname_updates[ $class_name ] = self::REMOVE_CLASS;
+			$this->classname_updates[ $this->comparable_class_name( $class_name ) ] = self::REMOVE_CLASS;
 		}
 
 		return true;
 	}
 
+	/**
+	 * Transform a class name string to a comparable form.
+	 *
+	 * This method may be subclassed to customize class names for comparison. For example, this
+	 * allows for subclasses to support case-insensitive class name comparison.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @param string $class_name The class name to transform.
+	 * @return string The transformed class name.
+	 */
+	protected function comparable_class_name( string $class_name ): string {
```

Comment:
why not create this as an internal method and add the quirks-mode flag to the Tag Processor?

Reply:
That would simplify some things, happy to make that change if you prefer.

I didn't add it because it's an HTML concept and seems more appropriate for the HTML processor. Those lines have already started to blur.

Reply:
in this case I see this more as a CSS concept and I don't see a reason to make the Tag Processor less adequate. CSS class name matching was always a concept we incorporated because of the obvious need to use it. the CSS class name matching has always had custom code.

I would rather in this case embed the quirks mode check in the Tag Processor, or at least assume _no-quirks_ mode by default in the absence of that. This _does_ imply that the quirks mode flag should live in the Tag Processor, but that seems reasonable since normal HTML parsing won't examine it.